rs/sns/testing/src/main.rs

```diff
@@ -37,38 +69,88 @@ async fn main() {
-    let dev_participant_id = PrincipalId::new_user_test_id(1000);
+    let dev_principal_id = get_identity_principal(&opts.dev_identity).unwrap();
+    let treasury_principal_id = get_identity_principal(&opts.treasury_identity).unwrap();
 
     bootstrap_nns(
         &pocket_ic,
         vec![initial_mutations],
-        vec![],
-        vec![dev_participant_id],
+        vec![
+            (
+                treasury_principal_id.into(),
+                Tokens::from_tokens(10_000_000).unwrap(),
+            ),
+            (dev_principal_id.into(), Tokens::from_tokens(100).unwrap()),
+        ],
+        vec![dev_principal_id],
     )
     .await;
+}
 
-    let greeting = "Hello there".to_string();
-    let test_canister_id = install_test_canister(
-        &pocket_ic,
-        TestCanisterInitArgs {
-            greeting: Some(greeting),
-        },
+async fn run_basic_scenario(opts: RunBasicScenarioOpts) {
+    let dfx_interface = dfx_interface(&opts.network, Some(opts.dev_identity))
+        .await
+        .unwrap();
+
+    let network = dfx_interface.network_descriptor();
+
+    let dev_agent = dfx_interface.agent();
+
+    let treasury_agent = &get_agent(&opts.network, Some(opts.treasury_identity))
+        .await
+        .unwrap();
+
+    let seed = 322_u64;
+    let mut rng = ChaChaRng::seed_from_u64(seed);
+
+    let swap_participants_keys = (0..20).map(|_r| SecretKey::random(&mut rng));
+    let swap_participants_agents = join_all(
+        swap_participants_keys
+            .map(|k| async { build_ephemeral_agent(k, &network.clone()).await.unwrap() }),
     )
     .await;
-    println!("Test canister ID: {}", test_canister_id);
+
+    let nns_neuron_hotkeys = get_nns_neuron_hotkeys(dev_agent, NNS_NEURON_ID)
+        .await
+        .unwrap();
+
+    assert!(
+        nns_neuron_hotkeys.contains(&dev_agent.caller().unwrap().into()),
+        "Developer identity is not a hotkey for NNS neuron"
+    );
+
     println!("Creating SNS...");
-    let sns = create_sns_pocket_ic(&pocket_ic, dev_participant_id, vec![test_canister_id]).await;
+    let sns = create_sns(
+        dev_agent,
+        NNS_NEURON_ID,
+        dev_agent,
+        treasury_agent,
+        swap_participants_agents,
+        vec![opts.test_canister_id],
+    )
+    .await;
     println!("SNS created");
     println!("Upgrading SNS-controlled test canister...");
-    upgrade_sns_controlled_test_canister_pocket_ic(
-        &pocket_ic,
-        dev_participant_id,
+    upgrade_sns_controlled_test_canister(
+        dev_agent,
         sns,
-        test_canister_id,
+        opts.test_canister_id,
         TestCanisterInitArgs {
             greeting: Some("Hi".to_string()),
         },
     )
     .await;
-    println!("Test canister upgraded");
+    println!("Test canister upgraded")
+}
+
+#[tokio::main]
+async fn main() {
+    let opts = SnsTestingOpts::parse();
+
+    let subcommand = opts.subcommand;
```

Comment:
Does this variable/assignment really need to exist?

Reply:
Good point, removed